### PR TITLE
Adds a scripts to produce the downloadable zip files

### PR DESCRIPTION
**Pull request by ignasi35**
Supersedes #188 
References #61
Blocked by #225

### CODE REVIEW

File: scripts/prepare-dowloads.sh [hunk 1]
```diff
@@ -0,0 +1,21 @@
+#!/usr/bin/env bash
+
+declare -r tutorial_sources="docs-source/docs/modules/microservices-tutorial/examples"
+declare -r tutorial_attachments="docs-source/docs/modules/microservices-tutorial/assets/attachments"
+
+function prepareAttachment() {
+   source_name=$1
+   zip ${tutorial_attachments}/${source_name}.zip ${tutorial_sources}/${source_name}
```
Comment (patriknw) [inline scripts/prepare-dowloads.sh:8]: when unzipping the folder structure should be without the number prefixes
```
shopping-cart-service/
shopping-analytics-service/
shopping-order-service/
```

File: scripts/prepare-dowloads.sh [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+}
+
+mkdir -p ${tutorial_attachments}
+
+prepareAttachment 00-shopping-cart-service-scala
+prepareAttachment 01-shopping-cart-service-scala
+prepareAttachment 02-shopping-cart-service-scala
+prepareAttachment 03-shopping-cart-service-scala
+prepareAttachment 04-shopping-cart-service-scala
+prepareAttachment 05-shopping-cart-service-scala
```
Comment (patriknw) [inline scripts/prepare-dowloads.sh:18]: 05 is the result of "Projection publishing to Kafka". That zip should also include the shopping-analytics-service

File: scripts/prepare-dowloads.sh [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+prepareAttachment shopping-cart-service-scala
+prepareAttachment shopping-analytics-service-scala
+prepareAttachment shopping-order-service-scala
```
Comment (patriknw) [inline scripts/prepare-dowloads.sh:21]: There should be one zip with the full example, i.e. including all 3
```
shopping-cart-service/
shopping-analytics-service/
shopping-order-service/
```

**Comment by ignasi35** [comment]
Given the comments above, I think the approach I took here is wrong. Instead, I will create a temporal folder with the correct name (not prefixed, e.g. `01-`) where I'd copy the necessary contents (could be a mix of `05-shopping` and `analytics` for example) and then zip that temporal folder and move it to `assets/attachments`.

**Comment by patriknw** [comment]
Sounds good. Then we can also remote the Antora tags there before zipping.

**Comment by ignasi35** [comment]
This is ready for review.

I haven't worked on a solution to `remove the Antora tags there before zipping`. I am thinking something based on `sed` but I know it has some syntax differences between macOS and linux. I'm open to suggestions. I even think we can do that tag removal bit on a separate PR.

**Comment by patriknw** [comment]
> I am thinking something based on sed but I know it has some syntax differences between macOS and linux.

Yes, whatever works. If it only works on Linux (Travis) I'm fine with that too.

> I even think we can do that tag removal bit on a separate PR.

Yes

**Comment by ignasi35** [comment]
I've spiked the tag removal in https://travis-ci.com/github/akka/akka-platform-guide/builds/188417741#L226-L234 and it turned out to be simpler than I anticipated.

File: scripts/prepare-dowloads.sh [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+mkdir -p ${tutorial_attachments}
+
+## Remove Antora tags from codebase
+function removeTags() {
+   pushd ${tutorial_sources}
+   find . -type f -print0 | xargs -0 sed -i "s/\/\/ tag::[^\[]*\[.*\]//g" 
+   find . -type f -print0 | xargs -0 sed -i "s/# tag::[^\[]*\[.*\]//g" 
+   find . -type f -print0 | xargs -0 sed -i "s/\/\/ end::[^\[]*\[.*\]//g" 
+   find . -type f -print0 | xargs -0 sed -i "s/# end::[^\[]*\[.*\]//g"
```
Comment (patriknw) [inline scripts/prepare-dowloads.sh:16]: could we also remove `// <1>` callouts?

File: scripts/prepare-dowloads.sh [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+}
+
+## Remove the tags used by Antora snippets from 
+## the codebase before zipping
+removeTags
+
+
+## gRPC service
+prepareTemporalFolder
+fetchProject 01-shopping-cart-service-scala shopping-cart-service
```
Comment (patriknw) [inline scripts/prepare-dowloads.sh:54]: I think we can create one for 00 also. g8 is the recommended seed for Scala but if that doesn't work for some reason they can download 00 when starting at the gRPC service.

**Review by patriknw — COMMENTED** [review]
looking great

File: scripts/prepare-dowloads.sh [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+   ## remove tags from code
+   find . -type f -print0 | xargs -0 sed -i "s/\/\/ tag::[^\[]*\[.*\]//g" 
+   find . -type f -print0 | xargs -0 sed -i "s/\/\/ end::[^\[]*\[.*\]//g" 
+   
+   ## remove tags from config
+   find . -type f -print0 | xargs -0 sed -i "s/# tag::[^\[]*\[.*\]//g" 
+   find . -type f -print0 | xargs -0 sed -i "s/# end::[^\[]*\[.*\]//g" 
+
+   ## remove call-outs
+   find . -type f -print0 | xargs -0 sed -i "s/\/\/ <[0-9]*>//g"
```
Comment (ignasi35) [inline scripts/prepare-dowloads.sh:23]: This is the only regexp I didn't test.

File: scripts/prepare-dowloads.sh (hunk 5 above)
Comment (octonato) [reply scripts/prepare-dowloads.sh:68]: Do we have any other reason for having a 00 variant? 

If the only reason is that .g8 may not work, then I don't think we should have it at all. 

That's yet another thing to keep in sync for us.

File: scripts/prepare-dowloads.sh [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+
+## Remove the tags used by Antora snippets from 
+## the codebase before zipping
+removeTags
+
+
+## empty codebase (this is a backup option for users 
+## that can't use the .g8 template)
+prepareTemporalFolder
+fetchProject 00-shopping-cart-service-scala shopping-cart-service
```
Comment (octonato) [inline scripts/prepare-dowloads.sh:63]: there is no `00-shopping-cart-service-scala`. @ignasi35, maybe you forgot to `git add .`?

File: scripts/prepare-dowloads.sh [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+zipAndAttach step-2.zip
```
Comment (octonato) [inline scripts/prepare-dowloads.sh:64]: Maybe better if we don't number the zips and give it a meaningful name. Should that be easier for users to link back to tutorial step?

File: scripts/prepare-dowloads.sh (hunk 7 above)
Comment (ignasi35) [reply scripts/prepare-dowloads.sh:63]: This PR must be merged after the one that splits the code.

There is no 04 and 05 either.  :-)

File: scripts/prepare-dowloads.sh (hunk 8 above)
Comment (ignasi35) [reply scripts/prepare-dowloads.sh:64]: Sure thing.
Let's do that when adding the links to the assets (separate PR).

**Review by patriknw — APPROVED** [review]
LGTM

**Pull request merged**